sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,164 +1,122 @@
-There is also an explicit cast operator that can be used to convert the strongly typed model to the protocol RequestContent.
+## Key Concepts
 
-### Serialization
+- [Using the ModelSerializer](#using-the-modelserializer)
+- [Using models with protocol methods](#using-models-with-protocol-methods)
+- [Using JsonSerializer](#using-jsonserializer)
+- [Envelope bring your own model case](#envelope-bring-your-own-model-case)
 
-```C# Snippet:ExplicitCast_Serialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-DogListProperty dog = new DogListProperty("myPet");
-Response response = client.CreatePet("myPet", (RequestContent)dog);
-var response2 = client.CreatePet("myPet", RequestContent.Create(dog));
-```
-
-### Deserialization
-
-```C# Snippet:ExplicitCast_Deserialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-Response response = client.GetPet("myPet");
-DogListProperty dog = (DogListProperty)response;
-Console.WriteLine(dog.IsHungry);
-```
+## Using the ModelSerializer
+The `ModelSerializer` class enables users to serialize and deserialize any Azure models that implement the `IModelSerializable<T>` interface. The following example shows a Dog model being serialized and deserialized.
 
-Given that explicit cast does not allow for serialization options we might also consider a static `FromResponse` and instance `ToRequestContent` methods.
+```C# Snippet:BaseModelSerializer
+Dog doggo = new Dog
+{
+    Name = "Doggo",
+    Age = 7
+};
 
-## Using ModelSerializer for NewtonSoftJson
+BinaryData data = ModelSerializer.Serialize(doggo);
 
-By using the ModelSerializer class, a new instance of Dog does not need to be created before calling Deserialize. Also added ObjectSerializer to Options class so different kinds of Serializers can be used.
+Dog dog = ModelSerializer.Deserialize<Dog>(data);
+```
 
-### Serialization
+By default, the ModelSerializer gives you the Json representation of the model. Some services accept XML so if you want to send the serialized data to an Azure service that accepts XML, you can use the `ModelSerializerFormat` enum to specify the format. By using the `Wire` format, the serializer automatically picks what the service would use.
 
-```C# Snippet:NewtonSoft_Serialize
-DogListProperty dog = new DogListProperty
+```C# Snippet:ModelSerializerWithFormat
+Dog doggo = new Dog
 {
     Name = "Doggo",
-    IsHungry = true,
-    Weight = 1.1,
-    FoodConsumed = { "kibble", "egg", "peanut butter" },
+    Age = 7
 };
-ModelSerializerOptions options = new ModelSerializerOptions();
-options.GenericTypeSerializerCreator = type => type.Equals(typeof(DogListProperty)) ? new NewtonsoftJsonObjectSerializer() : null;
-
-BinaryData data = ModelSerializer.Serialize(dog, options);
-```
 
-### Deserialization
+ModelSerializerOptions options = new ModelSerializerOptions(format: ModelSerializerFormat.Wire);
```

Comment:
Moot because we don't actually define `Dog` - but it sure would be confusing to me learning about this if I change to `Wire` format and the output looks exactly the same.   

Some additional discussion of what each format is and when they should be used may help, but we can't guarantee that developers will read it.   The snippets should work copy/paste and demonstrate.